draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2883,7 +2889,125 @@ https://video.example.com/k/movie1/4/013.ts
+		</t>
+
+
+		</list>
+
+	</t>
+	<t>
+	uCDN may combine multiple options in the same trigger command. dCDN should consider the following when processing such commands:
+	<list style="symbols">
+	<t>(i) dCDN MUST reject triggers that are dependent on other pending triggers with lower priority to prevent deadlocks.</t>
+	<t>(ii) When introducing an urgent trigger uCDN SHOULD indicate priority that is equal to or higher than the highest priority among the pending triggers that belong to the same uCDN. dCDN MUST reject urgent triggers that have lower priority than other pending triggers of the same uCDN.</t>
+	<t>(iii) dCDN MUST reject an urgent trigger that has pending triggers as its prerequisite. dCDN MAY reject an urgent trigger that has triggers in “active” or “cancelling” state as its prerequisite, for example when such triggers are deemed to take a long time to complete or to cancel.</t>
+	</list>
+	
+	</t>
+	<t>
+	Object specification:
+
+        <t><list style="empty" hangIndent="3">
+        <t><list style="hanging" hangIndent="3"><t hangText="Name: priority">
+        <vspace blankLines="1"/>
+	Description: Relative weight of the trigger. 
+	When picking a trigger for execution from all pending triggers posted by each uCDN, 
+	the dCDN MUST choose the trigger with the highest priority first. 
+        <vspace blankLines="1"/>
+	Value: Integer from 0 to 100.
+        <vspace blankLines="1"/>
+	Mandatory: No. The value defaults to zero if omitted.
+        </t>
+        </list></t>
+        </list></t>
+
+
+        <t><list style="empty" hangIndent="3">
+        <t><list style="hanging" hangIndent="3"><t hangText="Name: depends">
+        <vspace blankLines="1"/>
+	Description: Links to Trigger Status Resources that the current Trigger Command depends on.
+	Indicates which triggers should fully finish processing before starting execution of the current trigger. 
+	The triggers need to be in one of the following states to be considered finished: 
+	“complete”, “processed”, “failed” or “cancelled”.
+        <vspace blankLines="1"/>
+	Value: A JSON array of zero or more URLs, represented as JSON strings.
+        <vspace blankLines="1"/>
+	Mandatory: No.
+        </t>
+        </list></t>
+	</list></t>
+
+        <t><list style="empty" hangIndent="3">
+        <t><list style="hanging" hangIndent="3"><t hangText="Name: urgent">
+        <vspace blankLines="1"/>
+	Description: Indicates whether the trigger should be immediately moved to the “active” state upon acceptance. In absence of this flag, the dCDN is at liberty to choose the time for trigger execution, e.g., batch processing.
```

Comment:
Or, would it be more accurate to say "at the top of the pending queue" upon acceptance?  What if the dCDN processes purge / preposition triggers serially, and there is already an active job that needs to complete first?

Reply:
This is not really binding then, the active job may take hours to complete. 
If dCDN cannot move process the urgent trigger ASAP (for example complete the active trigger right away), it should indicate that back to uCDN and uCDN will have a choice to cancel the active trigger if the urgent trigger is really urgent.

Reply:
Good point, thanks.